src/error/record.rs

```diff
@@ -19,37 +21,43 @@ pub enum RecordErrorKind {
     EmptyAlias,
 }
 
-impl fmt::Display for RecordError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "Invalid key '{}': ", self.input)?;
+impl ShortError for RecordError {
+    fn short_err(&self) -> &'static str {
         match self.kind {
-            RecordErrorKind::EmptyProvider => {
-                f.write_str("provider must contain non-whitespace characters")
-            }
-            RecordErrorKind::EmptySubId => {
-                f.write_str("sub-id must contain non-whitespace characters")
-            }
-            RecordErrorKind::EmptyAlias => {
-                f.write_str("alias must contain non-whitespace characters")
-            }
-            RecordErrorKind::InvalidProvider => f.write_str("provider is invalid"),
-            RecordErrorKind::InvalidSubId => {
-                f.write_str("sub-id is invalid for the given provider")
-            }
-            RecordErrorKind::RecordIdIsNotAlias => f.write_str("alias must not contain a colon"),
-            RecordErrorKind::RecordIdIsNotRemoteId => f.write_str("remote id must contain a colon"),
+            RecordErrorKind::EmptyProvider => "provider must contain non-whitespace characters",
+            RecordErrorKind::EmptySubId => "sub-id must contain non-whitespace characters",
+            RecordErrorKind::InvalidProvider => "alias must contain non-whitespace characters",
+            RecordErrorKind::InvalidSubId => "provider is invalid",
+            RecordErrorKind::RecordIdIsNotAlias => "sub-id is invalid for the given provider",
+            RecordErrorKind::RecordIdIsNotRemoteId => "alias must not contain a colon",
+            RecordErrorKind::EmptyAlias => "remote id must contain a colon",
```

Comment:
The messages have been mixed up in the reordering.

Reply:
Fixed now! Good catch, I guess we should have tests to catch this, but we are missing tests for a lot of things more generally...